R/hy_db.R

```diff
@@ -109,5 +105,50 @@ hy_test_db <- function() {
   system.file("test_db/tinyhydat.sqlite3", package = "tidyhydat")
 }
 
+#' Set the default database path
```

Comment:
What is the application here? For vignettes I used [this](https://community.rstudio.com/t/precompiling-vignette-with-devtools/1583/6) approach. The solution involved setting a [global knitr option](https://raw.githubusercontent.com/ropensci/tidyhydat/master/vignettes/tidyhydat_an_introduction.Rmd) for `eval` as TRUE or FALSE depending on the presence of an environment variable. Locally when I build the vignettes, the environment variable is present so `eval=TRUE`. On CRAN, the environment variable is absent for `eval=FALSE`. Therefore the [vignette code becomes runnable](https://cran.r-project.org/web/packages/tidyhydat/vignettes/tidyhydat_an_introduction.html) on CRAN. I like this approach as it is a little cleaner by potentially providing users with functions that could completely mess with their ability to access hydat. If this is only for compiling vignettes I'd rather go with the above approach. But if this has other applications let me know.

Reply:
So to be clear, I am not advocating to remove this functionality. I think you've outlined a much more elegant way of accessing the test database. To address this:
> This has the advantage of your vignettes getting built every time you check the package, which will make it more apparent which future code changes break the vignette. 
The vignette is actually built every time I check the package because I have the environmental variable set. I don't think it matters too much and certainly not enough to matter here.